core/dive.c

```diff
@@ -277,27 +277,40 @@ struct event *peek_next_divemodechange(struct event *evd)
 	return (NULL);
 }
 
+/*
+enum dive_comp_type get_divemode_at_time(struct divecomputer *dc, int divetime, struct event **ev_dmc)
+{ // The dummy version of this function, call parameters compatible with self-tracking version.
+ return dc->divemode;
+}
+
+enum dive_comp_type get_divemode_at_time(struct divecomputer *dc, int divetime, struct event **ev_dmc)
+{ // The non-self-tracking version of this function, call parameters compatible with self-tracking version.
+	struct event *ev, *ev_ndc = dc->events;
+	enum dive_comp_type mode = dc->divemode;
+	ev = get_next_divemodechange(&ev_ndc);
+	while (ev && (divetime >= ev->time.seconds)) {
+		mode = ev->divemode;
+		ev = get_next_divemodechange(&ev_ndc);
+	}
+	return mode;
+} */
+
```

Comment:
i wonder if it's better to have this around `#ifdef 0`. are there plans to enable it in the future.
might in a next commit in the serious as i'm going one at a time.